### PR TITLE
[P2022-1723] add routing to issue details

### PR DESCRIPTION
**Pull request by jprymak**
[https://tracker.intive.com/jira/secure/RapidBoard.jspa?rapidView=1485&view=detail&selectedIssue=P2022-1723&sprint=4957](https://tracker.intive.com/jira/secure/RapidBoard.jspa?rapidView=1485&view=detail&selectedIssue=P2022-1723&sprint=4957)

- removed temporary issue component
- added routing to issue details
- added api integration - displaying issues in board columns

### CODE REVIEW

File: src/mockData/mockBoardStatus.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export const mockBoardStatus = [
-  { code: "To do", id: 1 },
-  { code: "in progress", id: 2 },
-  { code: "done", id: 3 },
+  { boardId: 1, statusId: 1, status: { code: "To do", id: 1 } },
+  { boardId: 2, statusId: 2, status: { code: "in progress", id: 2 } },
+  { boardId: 3, statusId: 3, status: { code: "done", id: 3 } },
```
Comment (Voland91) [inline src/mockData/mockBoardStatus.tsx:4]: I thinkt it should look like data from API, there is no boardId and statusId

File: src/api/requestsApi.ts [hunk 2]
```diff
@@ -20,28 +20,49 @@ const FetchDataAPI = {
...
 
-    const filteredBoardStatus = boardStatus.data.filter(
+    const boardStatusFilteredById = boardStatus.data.filter(
       (boardStatus: DataObject) => {
         return boardStatus.boardId == id;
       }
     );
 
-    if (filteredBoardStatus.length < 1) {
-      return [, status.data];
```
Comment (Voland91) [inline src/api/requestsApi.ts:30]: I need this status.data as a second parameter to correctly add new column

File: src/api/requestsApi.ts [hunk 3]
```diff
@@ -20,28 +20,49 @@ const FetchDataAPI = {
...
+          return boardStatusIds.includes(status.id);
+        }
+      );
+
+      const statusObject: DataObject = {};
+
+      statusFilteredByBoardStatusId.forEach((obj: any) => {
+        statusObject[obj.id] = obj;
       });
-      return [filteredStatus, status.data];
```
Comment (Voland91) [inline src/api/requestsApi.ts:42]: Same situation like in line 30

Comment (jprymak) [reply src/api/requestsApi.ts:42]: I am a little but confused on this one :)

File: src/mockData/mockBoardStatus.tsx (hunk 1 above)
Comment (jprymak) [reply src/mockData/mockBoardStatus.tsx]: I've changed the code resemble data returned from api

**Review by Voland91 — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -39,8 +22,10 @@ const App = () => {
             path='projects/:projectName&:projectId/:board&:boardId'
             element={<Board />}
           />
-          <Route path={toIssue()} element={<Issue />} />
-          <Route path='issue-details' element={<IssueDetails />} />
+          <Route
+            path='projects/:projectName&:projectId/:board/:issue&:issueId'
```
[the remark this answers is not in the document]
Comment (jprymak) [reply src/App.tsx]: done

File: src/api/requestsApi.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jprymak) [reply src/api/requestsApi.ts:30]: done

**Pull request merged**